task_manager_using_typescript/src/interfaces.ts

```diff
@@ -0,0 +1,12 @@
+export interface InterfaceTask {
+  id: string;
+  name: string;
+  isCompleted: boolean;
+  isFavorite: boolean;
+}
+
+export interface InterfaceAllTasks {
+  name: string;
+  isCompleted: boolean;
+  isFavorite: boolean;
+}
```

Comment:
You can extend the `InterfaceAllTasks ` with `InterfaceTask` as some most of the fields are redundant